src/all/mangapark/src/eu/kanade/tachiyomi/extension/all/mangapark/MangaParkDto.kt

```diff
@@ -96,6 +126,14 @@ class MangaParkComic(
             }
             return result.toString()
         }
+
+        private fun useLatestPageAsCover(genres: List<String>?): Boolean {
+            return genres.orEmpty().let {
+                it.contains("hentai") && !it.contains("webtoon")
+            }
+        }
+
+        private val shortenTitleRegex = Regex("""(\[[^]]*]|[({][^)}]*[)}])""")
```

Comment:
Why make it so a `(` can be closed by `}`? Consider using `RegexOptions.COMMENTS`, then you can use (and add comments):
```
(
    \(         # Brackets: (...)
        [^)]*
    \)
    |          # OR
    \[         # Brackets: [...]
        [^]]*
    \]
    |          # OR
    \{         # Brackets: {...}
        [^}]*
    \}
)
```

You _might_ be able to replace the inner regex with `.*?`, as it will still need to match the closing bracket. This would help making the regex less complex.

Reply:
There might be some additional improvements for cleaning titles. I have categorized some titles observed from the site into [a gist](https://gist.github.com/vetleledaal/792e3a420456014688850932e9a9fa19). Of course subjective, but you are free to make your own conclusion.

---

With the stated goal from the gist and to minimize false positives I think it would be better if you have something where:
- the regex only matches the start OR end of a title
- Recursively:
  - remove match
  - trim (or match `\s` in regex)

Reply:
https://regex101.com/r/Ww8YNd/1